File: ruby/raindrops/raindrops.rb
module Raindrops
  SOUNDS = [[3, "Pling"], [5, "Plang"], [7, "Plong"]]

  def self.convert(n)
    sounds = SOUNDS.reduce("") { |s, (factor, sound)| n % factor == 0 ? s + sound : s }
    sounds.empty? ? n.to_s : sounds
  end
end

File: kotlin/raindrops/_iterations/2/src/main/kotlin/Raindrops.kt
object Raindrops {
    fun convert(input: Int): String {
        val sb = StringBuilder()
        if (input % 3 == 0) {
            sb.append("Pling")
        }
        if (input % 5 == 0) {
            sb.append("Plang")
        }
        if (input % 7 == 0) {
            sb.append("Plong")
        }
        if (sb.isEmpty()) {
            sb.append(input)
        }
        return sb.toString()
    }
}
